pkg/api/object_channel.go

```diff
@@ -0,0 +1,122 @@
+package api
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+
+	"github.com/anexia-it/go-anxcloud/pkg/api/types"
+	"github.com/go-logr/logr"
+)
+
+const (
+	// ListChannelDefaultPageSize specifies the default page size for List operations returning the data via channel.
+	ListChannelDefaultPageSize = 10
+)
+
+type objectChannel struct {
+	ctx       context.Context
+	ctxCancel context.CancelFunc
+
+	api API
+	log logr.Logger
+
+	objects      chan types.ObjectRetriever
+	abort        chan bool
+	pageIterator types.PageInfo
+}
+
+// NewObjectChannel creates a new object implementing types.ObjectChannelCloser which can be used after passed to API.List()
+// with the ObjectChannel option.
+func NewObjectChannel() types.ObjectChannelCloser {
+	return &objectChannel{
+		objects: make(chan types.ObjectRetriever),
+		abort:   make(chan bool),
+	}
+}
+
+// Channel starts the object listing logic and returns the channel objects can be retrieved from.
+func (oc *objectChannel) Channel() (types.ObjectChannel, error) {
+	if oc.pageIterator == nil {
```

Comment:
I'm not entirely sure if we want to have such a check. It makes sense to block using a channel just created with NewObjectChannel(), but might make it unnecessarily hard to use it in goroutines (like one goroutine initializing and listing it, while another makes the `List()` call - because queuing API calls or whatever?). Not sure if this is a relevant use case.

Reply:
I think this check is perfectly fine. When using this with goroutines you would probably just handover the `chan` and not the `objectChannel` object, since the Close function should be called from a coordinating higher goroutine.